query-builder.js

```diff
@@ -229,6 +243,104 @@ class QueryBuilder {
     return this._addCondition('EMPTYSTRING', '', ['string']);
   }
 
+  /**
+   * Adds new 'SAMEAS' condition
+   * @param {*} field string name of compared field
+   * @returns {this} this
+   */
+  isSameAs(field) {
+    return this._addCondition('SAMEAS', field);
+  }
+
+  /**
+   * Adds new 'NSAMEAS' condition
+   * @param {*} field string name of compared field
+   * @returns {this} this
+   */
+  isNotSameAs(field) {
+    return this._addCondition('NSAMEAS', field);
+  }
+
+  /**
+  * Adds new 'GT_FIELD' condition
+  *
+  * @param {string} field name of field to compare against
+  * @returns {this} this
+  */
+  greaterThanField(field) {
+    return this._addComparisonCondition(field, 'GT_FIELD');
+  }
+
+  /**
+  * Adds new 'GT_OR_EQUALS_FIELD' condition
+  *
+  * @param {string} field name of field to compare against
+  * @returns {this} this
+  */
+  greaterThanOrIsField(field) {
+    return this._addComparisonCondition(field, 'GT_OR_EQUALS_FIELD');
+  }
+
+  /**
+  * Adds new 'LT_FIELD' condition
+  *
+  * @param {string} field name of field to compare against
+  * @returns {this} this
+  */
+  lessThanField(field) {
+    return this._addComparisonCondition(field, 'LT_FIELD');
+  }
+
+  /**
+  * Adds new 'LT_OR_EQUALS_FIELD' condition
+  *
+  * @param {string} field name of field to compare against
+  * @returns {this} this
+  */
+  lessThanOrIsField(field) {
+    return this._addComparisonCondition(field, 'LT_OR_EQUALS_FIELD');
+  }
+
```

Comment:
```suggestion

/**
 * @typedef {"year" | "month" | "day" | "hour" | "minute" | "second"} TimeUnit
 */

```

Reply:
This is interesting. What I find on running the `build:types` script, however, is that the described type is inlined. Where I documented the applicable types you will notice that they are not all represented in any given location. This reflects my experience testing the various units of time against a SNOW instance (I found the documentation on these conditions to be quite sparse; if you know of a better resource that would be great). If `TimeUnit` would be "promoted" to a named type I would feel okay about it, but if it is to be inlined onto each method then to me that represents a problem for the generated .d.ts to explicitly include a parameter value that, so far as I was able to find, does not work.

Reply:
Please consider commit 7735643c2fc2cc9bb3e53ae012003cdabbc913bc .

Reply:
I trust you on this. I changed jobs and no longer use ServiceNow day to day. So I can't really test this myself.